libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/marker/GeocodingComponent.kt

```diff
@@ -0,0 +1,44 @@
+package com.mapbox.navigation.dropin.component.marker
+
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.dropin.component.destination.DestinationAction.DidReverseGeocode
+import com.mapbox.navigation.dropin.component.destination.DestinationViewModel
+import com.mapbox.navigation.dropin.lifecycle.UIComponent
+import com.mapbox.navigation.dropin.util.Geocoder
+import com.mapbox.navigation.utils.internal.logW
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.mapNotNull
+
+/**
+ * UIComponent that observes and reverse geocodes Destination.
+ */
+internal class GeocodingComponent(
+    private val destinationViewModel: DestinationViewModel
+) : UIComponent() {
+
+    override fun onAttached(mapboxNavigation: MapboxNavigation) {
+        super.onAttached(mapboxNavigation)
+
+        val accessToken = checkNotNull(mapboxNavigation.navigationOptions.accessToken) {
```

Comment:
checkNotNull throws an IllegalStateException if the value is null.  The code as is would result in a crash if the token is null.  Please change this to use either ifNonNull or an alternative that can't result in a thrown exception.

Reply:
https://github.com/mapbox/mapbox-navigation-android/blob/656204930b17bc437e57b4d8bedefd17bda74816/libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/audioguidance/AudioGuidanceServices.kt#L27

I actually added a `!!` assuming we would have an access token. 

Figured we only need to check for accessToken in one place for drop in ui
https://github.com/mapbox/mapbox-navigation-android/blob/656204930b17bc437e57b4d8bedefd17bda74816/libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/DropInNavigationView.kt#L172

Reply:
Thanks @cafesilencio I am aware of that. This expected. We should never pass MapboxNavigation instance that doesn't have `mapboxNavigation.navigationOptions.accessToken`. Throwing IllegalStateException here will ensure that we never do.

Reply:
Throwing exceptions crashes our customers' applications.  We should refrain from implementing code that can crash the application using our SDK.  It's easy enough to be defensive with nullables.

Reply:
@abhishek1508 How do you want to handle this?

Reply:
I just fixed a bug caused by using this kind of checking: https://github.com/mapbox/mapbox-navigation-android/pull/5524.  It isn't this class's responsibility to validate the access token. From this class's point of view it's accessing a nullable variable.

Reply:
We can just use `!!` as the drop in ui won't work without the access token and we are already failing it early.

Reply:
Using `!!` will crash the system if used to force unwrap `null`. 

Technically both implementations lead to the same result, exception when accessToken is null.
The difference is that the former approach is more explicit about the check and will throw `IllegalStateException`, while latter is implicit and will throw NPE.

```kotlin
val accessToken = checkNotNull(mapboxNavigation.navigationOptions.accessToken) {
  "Missing AccessToken"
}
```

```kotlin
val accessToken = mapboxNavigation.navigationOptions.accessToken!!
```